docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+`{type} {variable name} = {value};`.
+
+#### 5.1 Data Types
+
+#### 5.1.1. `int`
+
+Identifiers of type `int` represent the natural numbers. The upper and lower bounds for int are defined by the architechtural constraints of the computer.
+
+Example:
+
+`int num = 32;`
+
+`int` types can also be assigned to the result of expressions:
+
+`int a = 34 * 2 + (2 / 1);`
+
+#### 5.1.2. `str`
+
+Identifiers of type `str` are used to represent sequences of characters, strings. Strings are immutable and can be declared in the following manner:
+
+Example:
+
+`str myString = "Hello World";`
+
+#### 5.1.3. `file`
+
+Files are regarded as first-class citizens in FIRE. This is made apparent by the importance and centrality of files. A `file` type represents either an existing file or a file that is to be written. This allows the programmer to more easily perform operations on the file.
+
+The syntax for instantiating a `file` object is as follows:
+
+`file f = file("filename.csv", "<mode>");`
+
+In the example provided above, two arguments are fed into the `file(...)` argument: *filename* for reading, writing or both, and *mode*. The *mode* argument can be `r` for read only, `w` for write only, and `rw` for both.
+
+Example:
+
+`file f = file("test.csv", "rw");` will open the File named test.csv in the current directory for both reading and writing.
+  
+`file f = file("filename.csv", "<mode>", "<delim>");`
+
+An optional third argument *delim* may be provided to the constructor specifying a delimiter for reading. If the *delim* argument is not supplied it will default to `\n`.
+
+Example:
+
+`file f = file("Program.java", "rw", ";");` will open the File named `Program.Java` in the current directory for both reading and writing. Calls to `read()` will read in chunks of the file delimited by the `;` character.
+
+#### 5.1.4. `func`
+
+`func` objects reference functions and are treated as first class citizens. The structure of `func` variable declarations is as follows:
+
+`func <return type> <name> = (<parameters>) => { <function body> };`
+
+Where:
+ * `<return type>` is the type returned by the function
+ * `<name>` is the variable name of the function
+ * `<paramters>` are the expected parameters for the function
+ * `<function body>` is the body of the function
+ 
+Once a function has been assigined to a `func` type it becomes a "named function" that is callable using that name e.g `funcName();`
+
+A function that does not return anything has a return type of `void`.  The void return type allows for programmers to create functions that are useful for their side effects. 
+
+##### Paramaterization
+
+Named functions can be passed to other functions as a parameter as follows\:
+```
+func void saySomething = () =>{ print("something"); };
+func void doSomething = (func f) => { f(); };
```

Comment:
how do you handle the following (which does not type check):

```
func void saySomething = (str s) => { print(s); } ;
func void doSomething = (func f) => { f(); } ;
doSomething(saySomething);
```

Reply:
you claim to not have genericity/polymorphism, yet as far as typing is concerned, `func f` implies that `f` can be of any function type.